zeeguu_api/api/user_working_session.py

```diff
@@ -0,0 +1,173 @@
+@cross_domain
+@with_session
+def working_sessions_by_user():
+    """
+        Returns user_working_sessions data based on the provided filters for the specified user
+
+        Parameters:
+            user_id
+            from_date   (using ISO-8601 format)
+            to_date     (using ISO-8601 format)
+            is_active
+
+        Example: post
+        http://localhost:9001/working_sessions_by_user?user_id=534&from_date=2017-02-24T00:00:00&to_date=2018-02-24T09:13:13&is_active=False&session=92752705
+       
+        return:
+        [[session1], [session2], ... [sessionn]]
+        where session
+        [session_id, start_time, article_id, duration (in miliseconds)]
+        Example:
+        [
+            [
+                147,
+                "2018-02-24T09:11:11",
+                null,
+                61
+            ],
+            ...
+        ]
+    """
+    user_id =  int(flask.request.args['user_id'])
+    from_date = flask.request.args['from_date']
+    to_date = flask.request.args['to_date']
+    is_active = bool(flask.request.args['is_active'])
+    active_sessions = UserWorkingSession.find_by_user(user_id, from_date, to_date, is_active)
+    result = [[item.id, item.start_time.isoformat(), item.article_id, item.duration] for item in active_sessions]
+    return json_result(result)
+
+
+# ---------------------------------------------------------------------------
+@api.route("/working_sessions_by_cohort", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def working_sessions_by_cohort():
+    """
+        Returns user_working_session data based on the provided filters for the specified cohort
+
+        Parameters:
+            cohort_id
+            from_date   (using ISO-8601 format)
+            to_date     (using ISO-8601 format)
+            is_active
+
+        Example: post
+        http://localhost:9001/working_sessions_by_cohort?cohort_id=1&from_date=2010-01-24T00:00:00&to_date=2018-02-24T09:13:13&is_active=False&session=92752705
+        
+        return:
+        [[session1], [session2], ... [sessionn]]
+        where sessionn
+        [user_working_session.id, user_id, start_time, article_id, duration (in miliseconds)]
+        Example:
+        [
+            [
+                48,
+                617,
+                "2017-05-19T14:58:58",
+                63399,
+                0
+            ],
+            ...
+        ]
+    """
+    cohort_id =  int(flask.request.args['cohort_id'])
+    from_date = flask.request.args['from_date']
+    to_date = flask.request.args['to_date']
+    is_active = bool(flask.request.args['is_active'])
+    active_sessions = UserWorkingSession.find_by_cohort(cohort_id, from_date, to_date, is_active)
+    result = [[item.id, item.user_id, item.start_time.isoformat(), item.article_id, item.duration] for item in active_sessions]
```

Comment:
actually, both here and in the previous example we could have returned a dictionary instead of an array for increased readability of the client-side code. if we return a dictionary the client can write code like: `result['id']`, `result['article_id']` as opposed to `result[0]` and `result[2]`. 

what do you think?